### DISCUSSION TITLE
Slim Application Error after (unsucessful) backup which consumed all disk space

### DISCUSSION THREAD

**Discussion by kristallhawaii** [nextcloud/all-in-one#3260] [Questions] [answered]
Hi,

**Host system**
Host-OS: debian 12 amd64
on an VPS which is only running NextcloudAIO
I have a separate partition which is encrypted with LUKS and using BTRFS.
That volume is mounted to /var/lib/docker, so everything from docker is on that partition
df -h
/dev/mapper/data-crypt  352G  260G   91G  75% /var/lib/docker
/dev/mapper/data-crypt  352G  260G   91G  75% /backup

**Steps I did**
At initial setup I set /backup as the backup-path, which also points to a subvolume of my "data volume".

I started the backup via AIO. After a while I checked the AIO webinterface and it seemed that, backup was finished. 
I hit the "start containers" button and then the " Slim Application Error" was shown.
I checked the shell and could see that the backup consumed all diskspace, because I didn't realize that my data is already this big. I had to delete the borg folder in /backup to get free space again.
But even after restart the container and the VPS and also a "docker rm nextcloud-aio-mastercontainer" and recreate it (see https://github.com/nextcloud/all-in-one/discussions/784#discussioncomment-2886870) didn't work.

I now was able to start all the other containers manually so I can continue use my nextcloud instance. But I guess something was broken while backup 

Thanks in advance

<details><summary>docker inspect nextcloud-aio-mastercontainer</summary>
<p>

`
`If your server has port 80 and 8443 open and you point a domain to your server, you can get a valid certificate automatically by opening the Nextcloud AIO Interface via:
https://your-domain-that-points-to-this-server.tld:8443
[Sun Aug 27 14:09:28 2023] PHP 8.2.8 Development Server (http://127.0.0.1:9876) started
[Sun Aug 27 14:09:28.577911 2023] [mpm_event:notice] [pid 101:tid 139687960636232] AH00489: Apache/2.4.57 (Unix) OpenSSL/3.1.2 configured -- resuming normal operations
[Sun Aug 27 14:09:28.577997 2023] [core:notice] [pid 101:tid 139687960636232] AH00094: Command line: 'httpd -D FOREGROUND'
{"level":"info","ts":1693145368.6089942,"msg":"using provided configuration","config_file":"/Caddyfile","config_adapter":""}
[27-Aug-2023 14:09:28] NOTICE: fpm is running, pid 106
[27-Aug-2023 14:09:28] NOTICE: ready to handle connections

[Sun Aug 27 14:09:45 2023] 127.0.0.1:57012 Accepted
[Sun Aug 27 14:09:45 2023] cloud.mrk-technik.de was accepted as valid domain.
[Sun Aug 27 14:09:45 2023] 127.0.0.1:57012 Closing
NOTICE: PHP message: Slim Application Error
Type: TypeError
Code: 0
Message: AIO\Data\ConfigurationManager::GetConfig(): Return value must be of type array, null returned
File: /var/www/docker-aio/php/src/Data/ConfigurationManager.php
Line: 15
Trace: #0 /var/www/docker-aio/php/src/Data/ConfigurationManager.php(360): AIO\Data\ConfigurationManager->GetConfig()
#1 /var/www/docker-aio/php/src/Controller/DockerController.php(235): AIO\Data\ConfigurationManager->GetDomain()
#2 /var/www/docker-aio/php/public/index.php(81): AIO\Controller\DockerController->StartDomaincheckContainer()
#3 /var/www/docker-aio/php/vendor/slim/slim/Slim/Handlers/Strategies/RequestResponse.php(38): Closure->{closure}(Object(GuzzleHttp\Psr7\ServerRequest), Object(GuzzleHttp\Psr7\Response), Array)
#4 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/Route.php(358): Slim\Handlers\Strategies\RequestResponse->__invoke(Object(Closure), Object(GuzzleHttp\Psr7\ServerRequest), Object(GuzzleHttp\Psr7\Response), Array)
#5 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(65): Slim\Routing\Route->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#6 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(65): Slim\MiddlewareDispatcher->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#7 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/Route.php(315): Slim\MiddlewareDispatcher->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#8 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/RouteRunner.php(68): Slim\Routing\Route->run(Object(GuzzleHttp\Psr7\ServerRequest))
#9 /var/www/docker-aio/php/vendor/slim/csrf/src/Guard.php(476): Slim\Routing\RouteRunner->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#10 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(168): Slim\Csrf\Guard->process(Object(GuzzleHttp\Psr7\ServerRequest), Object(Slim\Routing\RouteRunner))
#11 /var/www/docker-aio/php/vendor/slim/twig-view/src/TwigMiddleware.php(115): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#12 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(121): Slim\Views\TwigMiddleware->process(Object(GuzzleHttp\Psr7\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#13 /var/www/docker-aio/php/src/Middleware/AuthMiddleware.php(38): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#14 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(269): AIO\Middleware\AuthMiddleware->__invoke(Object(GuzzleHttp\Psr7\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#15 /var/www/docker-aio/php/vendor/slim/slim/Slim/Middleware/ErrorMiddleware.php(76): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#16 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(121): Slim\Middleware\ErrorMiddleware->process(Object(GuzzleHttp\Psr7\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
#17 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(65): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#18 /var/www/docker-aio/php/vendor/slim/slim/Slim/App.php(199): Slim\MiddlewareDispatcher->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#19 /var/www/docker-aio/php/vendor/slim/slim/Slim/App.php(183): Slim\App->handle(Object(GuzzleHttp\Psr7\ServerRequest))
#20 /var/www/docker-aio/php/public/index.php(181): Slim\App->run()
#21 {main}
Tips: To display error details in HTTP response set "displayErrorDetails" to true in the ErrorHandler constructor.
`
</p>
</details>

**Comment by szaimen** [comment]
Hi, which AIO version are you running?

**Reply by kristallhawaii** [reply]
Its ths latest, installed 7 days ago. I don't know how to get the actual AIO version. Nextcloud is Hub 5 27.0.2

**Reply by szaimen** [reply, marked as answer]
All right! Actually we have this check in place which should prevent the configuration.json file from being emptied: https://github.com/nextcloud/all-in-one/blob/7522d498180e08331b4c93112526b008b7b5337b/php/src/Data/ConfigurationManager.php#L516-L520

But afaik BTRFS sometimes has issues reporting the correct free size which would explain why the config file was emptied in your case (e.g. if it reports that space was still available but actually wasnt). Unfortunately I don't see how we can further improve this since this seem to be caused by using BTRFS in the first place.

**Comment by kristallhawaii** [comment]
Running out of space is always an issue.
Thank you for your fast answer and to fibd the issue!
But why is the config is touched during backup? Does this also containes states?

Anyway. How can I restore/recover the configuration.json? Recreation of the master container didn't do the trick

Thanks again!

**Reply by szaimen** [reply]
> How can I restore/recover the configuration.json?

Restore it from a backup
